File: Diyas DSA/q1 - Easy - Remove k.py
'''
You are given a string of characters
You must remove element k from the string 
It is not guarenteed that k will be in the string
Return the string without k
Refrain from using string.replace


EXAMPLE 1:
string = "hello world"
k = "l"

answer: heo word

EXAMPLE 2:
string = "I hate leetcode"
k = "e"

answer: I hat ltcod

Restrictions

0 < len(k) < 2
1 < len(TestString) < 50 
Testlist only contains lowercase letters
'''
# btw as a general note DO NOT rename the function given to you in the question.
def RemoveKelement(TestString:str,k:str) -> str:
    # Put your solution here 
    pass

















































































































































































import os
def check():
    testcases = [["i hate leetcode","e"],["hello world","l"],["coding club","z"],["python is great","t"],["mr walther","r"],["java sucks","a"],["the quick brown fox jumps over the lazy dog","o"]]
    for i in testcases:
        if mySolution(i[0],i[1]) != RemoveKelement(i[0],i[1]):
            return f'''
Failed Testcase {testcases.index(i)}

Testcase:
    TestString={i[0]}
    K={i[1]}

Your answer:
{RemoveKelement(i[0],i[1])}

Correct Answer:
{mySolution(i[0],i[1])}
'''
    if I_Told_You_I_Would_Check() == False:
        return "You passed!"
    else:
        return I_Told_You_I_Would_Check()

def mySolution(string,k):
    newstring = ""
    for i in string:
        if i != k:
            newstring += i
    return newstring

def I_Told_You_I_Would_Check() -> bool | str : 
    # I'm so typescriptcore
    with open(__file__,"r") as f:
        hold = f.readlines()
        # print(hold)
        i = 0
        # print ("import os" in hold[i] == False)
        while True:
            if ("import os" in hold[i]):
                break
            else:
                if ".replace(" in hold[i]:
                    return " \n Nice try nerd try the problem again without .replace() \n"
                i+=1
        return False

if __name__ == "__main__":
    print(check())

File: Diyas DSA/q1-Easy-RemoveK.cpp
#include <iostream>
#include <string>
#include <string.h>
#include <array>
using namespace std;

// You are given a string of characters
// You must remove element k from the string 
// It is not guarenteed that k will be in the string
// Return the string without k


// EXAMPLE 1:
// string = "hello world"
// k = "l"

// answer: heo word

// EXAMPLE 2:
// string = "I hate leetcode"
// k = "e"

// answer: I hat ltcode

// Restrictions

// 0 < (k).length() < 2
// 1 < (TestString).length() < 50 
// Testlist only contains lowercase letters

std::string mysolution(string str, string k);
std::string check();
std::string removeKelements(string str, string k);

std::string removeKelements(string TestString, string k) {
    // Implement your solution here
    return "";
}














































































































































































std::string check() {
    string testcases[7][2] = {{"i hate leetcode","e"},{"hello world","l"},{"coding club","z"},{"python is great","t"},{"mr walther","r"},{"java sucks","a"},{"the quick brown fox jumps over the lazy dog","o"}};
    // string testcases[][2] = {{"hello world","o"}};
    string temp;
    string tempk;
    string Asolution = "";
    string Bsolution = "";
    for (int i = 0; i < size(testcases); i++) {
        temp = testcases[i][0];
        tempk = testcases[i][1];
        Asolution = mysolution(temp,tempk);
        Bsolution = removeKelements(temp,tempk);
        if (Asolution != Bsolution) {
            return "you failed a testcase but I dont want to say which one because that would lead to a compiler error";
        };
    };
    return "You passed!";
};

std::string mysolution(string str, string k) {
    string newstring;
    string temp;
    for (int i = 0; i < size(str); i++) {
        temp = str[i];
        if (temp != k) {
            newstring.append( temp );
        }
    }
    return newstring;
}


int main() {
    std::cout << check() << std::endl;
}


// string TestStringCopy = TestString;
//     string temp;
//     int removedcharacters = 0;
//     for (int i = 0; i < size(TestString); i++) {
//         temp = TestString[i];
//         if (temp == k) {
//             TestStringCopy = TestStringCopy.erase(i-removedcharacters,1);
//             removedcharacters++;
//         };
//     };
//     return TestStringCopy;

// I'm just saving this because this entire program took me like 2 days :sob: